Lib/code.py

```diff
@@ -276,8 +315,20 @@ def raw_input(self, prompt=""):
         return input(prompt)
 
 
+class Quitter:
```

Comment:
Maybe just add a short docstring explaining that this is based on `_sitebuiltins.Quitter`, and how it is different.

Perhaps it's even worth just subclassing `Quitter` and overriding `__call__`? Not sure. These are both pretty simple, but also pretty tightly coupled. Your call.

Reply:
Subclassing `Quitter` won't give a lot of benefit, because the OS checking is outside the original `Quitter` class. We could calculate `eof` outside of the ``__init__`` function, it just seems like unnecessary. The class itself is already very compact and easy to understand.